realm/realm-library/src/objectServer/java/io/realm/SyncConfiguration.java

```diff
@@ -456,65 +530,101 @@ public ClientResyncMode getClientResyncMode() {
         @Nullable // null means the user hasn't explicitly set one. An appropriate default is chosen when calling build()
         private ClientResyncMode clientResyncMode = null;
         private long maxNumberOfActiveVersions = Long.MAX_VALUE;
+        private final BsonValue partitionKeyValue;
 
-        Builder(Context context, SyncUser user, String url) {
-            //noinspection ConstantConditions
+        /**
+         * FIXME
+         *
+         * @param user
+         * @param partitionValue
+         */
+        public Builder(RealmUser user, String partitionValue) {
+            this(user, new BsonString(partitionValue));
+        }
+
+        /**
+         * FIXME
+         *
+         * @param user
+         * @param partitionValue
+         */
+        public Builder(RealmUser user, ObjectId partitionValue) {
+            this(user, new BsonObjectId(partitionValue));
+        }
+
+        /**
+         * FIXME
+         *
+         * @param user
+         * @param partitionValue
+         */
+        public Builder(RealmUser user, int partitionValue) {
+            this(user, new BsonInt32(partitionValue));
+        }
+
+        /**
+         * FIXME
+         *
+         * @param user
+         * @param partitionValue
+         */
+        public Builder(RealmUser user, long partitionValue) {
+            this(user, new BsonInt64(partitionValue));
+        }
+
+        /**
+         * Builder used to construct instances of a SyncConfiguration in a fluent manner.
+         *
+         * @param user the user opening the Realm on the server.
+         * @param partitionValue the partition value specifying which subset of data to include in the Realm.
```

Comment:
this description is a bit ambiguous, maybe reformulate into something similar to https://github.com/realm/realm-java/pull/6788/files#diff-f19583eb27d992ab7655df81251f244cR489